tmk_core/common/action.h

```diff
@@ -79,16 +79,9 @@ extern bool disable_action_cache;
 #ifdef SWAP_HANDS_ENABLE
 extern bool                   swap_hands;
 extern const keypos_t PROGMEM hand_swap_config[MATRIX_ROWS][MATRIX_COLS];
-#    if (MATRIX_COLS <= 8)
-typedef uint8_t swap_state_row_t;
-#    elif (MATRIX_COLS <= 16)
-typedef uint16_t swap_state_row_t;
-#    elif (MATRIX_COLS <= 32)
-typedef uint32_t swap_state_row_t;
-#    else
-#        error "MATRIX_COLS: invalid value"
-#    endif
-
+#    ifdef ENCODER_MAP_ENABLE
+extern const uint8_t PROGMEM encoder_hand_swap_config[NUM_ENCODERS];
```

Comment:
This errors out, because it doesn't have `NUM_ENCODERS`, but even adding keymap.h doesn't help (it needs something else).  Moving this to action.c works though